include/modules/ircv3.h

```diff
@@ -45,7 +49,33 @@ class IRCv3::WriteNeighborsWithCap : public User::ForEachNeighborHandler
 		: cap(capability)
 		, protoev(ev)
 	{
-		user->ForEachNeighbor(*this, include_self);
+		sentid = user->ForEachNeighbor(*this, include_self);
+	}
+
+	already_sent_t GetAlreadySentId() const { return sentid; }
+};
+
+class IRCv3::WriteWatchersWithCap : public MonitorForEachWatcher::ForEachWatcherHandler
```

Comment:
It feels like mimicing the ForEachNeighbor code here is overengineering the problem. I'm not sure what would be an elegant way to implement it right now though.